packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -1174,6 +1204,276 @@ contract OPContractsManagerDeployer is OPContractsManagerBase {
+                _saltMixer: reusableSaltMixer(_input.opChainConfigs[0]),
+                _contractName: "DisputeGameFactory-Interop"
+            })
+        );
+
+        // Initialize the new DisputeGameFactory.
+        upgradeToAndCall(
+            _input.opChainConfigs[0].proxyAdmin,
+            address(newDisputeGameFactory),
+            getImplementations().disputeGameFactoryImpl,
+            abi.encodeCall(IDisputeGameFactory.initialize, (proxyAdminOwner))
+        );
+
+        // Deploy the new AnchorStateRegistry.
+        IAnchorStateRegistry newAnchorStateRegistry = IAnchorStateRegistry(
+            deployProxy({
+                _l2ChainId: block.timestamp,
+                _proxyAdmin: _input.opChainConfigs[0].proxyAdmin,
+                _saltMixer: reusableSaltMixer(_input.opChainConfigs[0]),
+                _contractName: "AnchorStateRegistry-Interop"
+            })
+        );
+
+        // Select the correct game type based on the type being used by the existing portals.
+        GameType newGameType;
+        if (_input.usePermissionlessGame) {
+            newGameType = GameTypes.SUPER_CANNON;
+        } else {
+            newGameType = GameTypes.SUPER_PERMISSIONED_CANNON;
+        }
+
+        // Initialize the new AnchorStateRegistry.
+        upgradeToAndCall(
+            _input.opChainConfigs[0].proxyAdmin,
+            address(newAnchorStateRegistry),
+            getImplementations().anchorStateRegistryImpl,
+            abi.encodeCall(
+                IAnchorStateRegistry.initialize,
+                (portals[0].superchainConfig(), newDisputeGameFactory, _input.startingAnchorRoot, newGameType)
+            )
+        );
+
+        // Migrate each portal to the new ETHLockbox and AnchorStateRegistry.
+        for (uint256 i = 0; i < portals.length; i++) {
+            // Authorize the existing ETHLockboxes to use the new ETHLockbox.
+            IETHLockbox existingLockbox = IETHLockbox(payable(address(portals[i].ethLockbox())));
+            newEthLockbox.authorizeLockbox(existingLockbox);
+
+            // Migrate the existing ETHLockbox to the new ETHLockbox.
+            existingLockbox.migrateLiquidity(newEthLockbox);
+
+            // Migrate the portal to the new ETHLockbox and AnchorStateRegistry.
+            portals[i].migrateToSuperRoots(newEthLockbox, newAnchorStateRegistry);
+        }
+
+        // Separate context to avoid stack too deep.
+        {
+            // Deploy a new DelayedWETH proxy for the permissioned game.
+            IDelayedWETH newPermissionedDelayedWETHProxy = IDelayedWETH(
+                payable(
+                    deployProxy({
+                        _l2ChainId: block.timestamp,
+                        _proxyAdmin: _input.opChainConfigs[0].proxyAdmin,
+                        _saltMixer: reusableSaltMixer(_input.opChainConfigs[0]),
+                        _contractName: "DelayedWETH-Interop-Permissioned"
+                    })
+                )
+            );
```

Comment:
Is there some reason we don't just keep the old delayed weth proxies? If nothing else, reusing existing ones keeps the state diff and gas costs down a bit.

Reply:
Unclear which ones to use. Each input chain has delayed weth proxies so you're forced to pick an input chain and use theirs. It'd probably be fine but seemed easier to just make new ones.

Reply:
Right these are interop set specific. In that case I agree with this approach.